pkg/csi_driver/reconciler.go

```diff
@@ -165,14 +174,349 @@ func (recon *multishareReconciler) reconcileWorker() {
+				// resize
+				klog.Infof("Starting share Resize operation for %s", shareURI)
+				_, err = recon.cloud.File.StartResizeShareOp(context.TODO(), share)
+			}
+		}
+		if err != nil {
+			recon.updateShareInfoErr(shareInfo, err)
+		}
+	}
+}
+
+func (recon *MultishareReconciler) maybeMarkShareInfoStatusDeleted(shareInfo *v1alpha1.ShareInfo, instanceInfo *v1alpha1.InstanceInfo) (*v1alpha1.InstanceInfo, error) {
+	if instanceInfo == nil {
+		return instanceInfo, fmt.Errorf("instanceInfo does not exist for shareInfo %s which is assigned to %q", shareInfo.Name, shareInfo.Status.InstanceHandle)
+	}
+
+	klog.Infof("mark shareInfo %s ShareStatus as deleted", shareInfo.Name)
+	instanceInfoClone := instanceInfo.DeepCopy()
+	instanceInfoClone, updated := recon.removeShareFromInstanceInfo(instanceInfoClone, shareInfo.Name)
+	if updated {
+		klog.Infof("trying to remove share %s from assigned instanceInfo %s", shareInfo.Name, instanceInfo.Name)
+		var err error
+		instanceInfoClone, err = recon.updateInstanceInfoStatus(context.TODO(), instanceInfoClone)
+		if err != nil {
+			return instanceInfo, fmt.Errorf("failed to remove share form instanceInfo Status: %s", err.Error())
+		}
+	}
+	shareInfoClone := shareInfo.DeepCopy()
+	if shareInfo.Status == nil {
+		shareInfoClone.Status = &v1alpha1.ShareInfoStatus{}
+	}
+	shareInfoClone.Status.ShareStatus = v1alpha1.DELETED
+	_, err := recon.updateShareInfoStatus(context.TODO(), shareInfoClone)
+	if err != nil {
+		return instanceInfoClone, fmt.Errorf("failed to update %s.Status.ShareStatus to DELETED: %s", shareInfo.Name, err.Error())
+	}
+
+	return instanceInfoClone, nil
+}
+
+func (recon *MultishareReconciler) sendInstanceRequests(instanceInfos map[string]*v1alpha1.InstanceInfo, ops []*Op) {
+	for _, instanceInfo := range instanceInfos {
+		needDelete := instanceInfo.DeletionTimestamp != nil
+		if !needDelete && instanceInfo.Status != nil &&
+			instanceInfo.Spec.CapacityBytes == instanceInfo.Status.CapacityBytes && instanceInfo.Status.InstanceStatus == v1alpha1.READY {
+			klog.V(6).Infof("no need to send any instance request for %s", instanceInfo.Name)
+			continue
+		}
+
+		instanceURI := util.InstanceInfoNameToInstanceURI(instanceInfo.Name)
+		op, err := runningOpMaybeErrForTarget(instanceURI, ops)
+		if err != nil {
+			recon.updateInstanceInfoErr(instanceInfo, err)
+		}
+		if op == nil {
+			klog.Infof("no running Op found for %s", instanceURI)
+			var instance *file.MultishareInstance
+			instance, err = recon.generateNewMultishareInstance(instanceInfo)
+			if err != nil {
+				klog.Errorf("error while generating instance for %s to call API: %s", instanceInfo.Name, err.Error())
+				continue
+			}
+
+			if needDelete {
+				// delete
+				klog.Infof("Starting instance Delete operation for %s", instanceURI)
+				_, err = recon.cloud.File.StartDeleteMultishareInstanceOp(context.TODO(), instance)
+			} else if instanceInfo.Status == nil || instanceInfo.Status.InstanceStatus != v1alpha1.READY {
+				// create
```

Comment:
these comments in L357, L361, L365 not needed

Reply:
done